components/director/internal/systemfetcher/service.go

```diff
@@ -354,19 +342,13 @@ func (s *SystemFetcher) processSystemsForTenant(ctx context.Context, tenantMappi
 				system.StatusCondition = app.Status.Condition
 			}
 
-			log.C(ctx).Infof("Started processing tenant business type for system with system number %s", systemNumber)
-			tenantBusinessType, err := s.processSystemTenantBusinessType(ctx, systemPayload, tenantBusinessTypes)
-			if err != nil {
-				return err
-			}
-
 			appInput, err := s.convertSystemToAppRegisterInput(ctx, system)
 			if err != nil {
 				return err
 			}
-			if tenantBusinessType != nil {
-				appInput.TenantBusinessTypeID = &tenantBusinessType.ID
-			}
+
+			log.C(ctx).Infof("Started processing tenant business type for system with system number %s", systemNumber)
+			appInput = s.enrichAppInputLabelsWithTenantBusinessType(systemPayload, appInput)
```

Comment:
The system fetcher e2e tests can be extended with enriching the system payload for the mocked systems with the two properties and adding a check for them to be present in the created application labels.